### PR TITLE
Api for retrieving eservice probing data

### PR DESCRIPTION
**Pull request by AlessandroColucci**

### CODE REVIEW

**Review by nttdata-rtorsoli — DISMISSED** [review]

File: src/main/java/it/pagopa/interop/probing/eservice/operations/util/EnumUtilities.java [hunk 1]
```diff
@@ -35,18 +35,28 @@ public List<String> convertListFromMonitorToPdnd(List<EserviceMonitorState> stat
...
     return switch (view.getState()) {
-      case ACTIVE -> checkND(view) ? EserviceMonitorState.N_D : EserviceMonitorState.ONLINE;
+      case ACTIVE -> checkND(view) ? EserviceMonitorState.N_D
+          : checkOFFLINE(view) ? EserviceMonitorState.OFFLINE : EserviceMonitorState.ONLINE;
       case INACTIVE -> checkND(view) ? EserviceMonitorState.N_D : EserviceMonitorState.OFFLINE;
       default -> throw new IllegalArgumentException("Invalid state {}= " + view.getState());
     };
   }
 
+  public boolean fromEnumToBoolean(EserviceInteropState viewState) {
```
Comment (galales) [inline src/main/java/it/pagopa/interop/probing/eservice/operations/util/EnumUtilities.java]: as someone that could use methods in this class, it's not clear what an api with signature `boolean EnumUtilities.fromEnumToBoolean(EserviceInteropState viewState)` does.

Maybe 
```suggestion
  public boolean isActive(EserviceInteropState viewState) {
```
?

File: docs/openapi/api-eservice-operations-interop-v1.yaml [hunk 2]
```diff
@@ -678,6 +723,22 @@ components:
         producerName:
           title: Service producer name
           type: string
+    
+    ProbingDataEserviceResponse:
+      type: object
+      properties:
+        probingEnabled:
```
Comment (galales) [inline docs/openapi/api-eservice-operations-interop-v1.yaml:743]: IMO this boolean is something related to the presentation layer, and should not be calculated here.
The `operations` layer should return data as close as possible to the persistence, doing transformations only when absolutely required (e.g. when querying by `EserviceMonitorState`)

File: docs/openapi/api-eservice-operations-interop-v1.yaml [hunk 3]
```diff
@@ -678,6 +723,22 @@ components:
...
+          title: probing activated
+          type: boolean
+        state:
+          $ref: '#/components/schemas/EserviceMonitorState'
+        responseReceived:
+          title: last response date
+          type: string
+          format: date-time
+        eserviceActive:
```
Comment (galales) [inline docs/openapi/api-eservice-operations-interop-v1.yaml]: same for presentation layer point

File: src/main/java/it/pagopa/interop/probing/eservice/operations/util/EnumUtilities.java (hunk 1 above)
Comment (AlessandroColucci) [reply src/main/java/it/pagopa/interop/probing/eservice/operations/util/EnumUtilities.java]: ok

File: docs/openapi/api-eservice-operations-interop-v1.yaml (hunk 2 above)
Comment (AlessandroColucci) [reply docs/openapi/api-eservice-operations-interop-v1.yaml:743]: There is already a technical debt in the backlog regarding moving the presentational logic from operations to probing-api.Maybe we can add this part to that?

File: docs/openapi/api-eservice-operations-interop-v1.yaml (hunk 3 above)
Comment (AlessandroColucci) [reply docs/openapi/api-eservice-operations-interop-v1.yaml]: same as above

File: docs/openapi/api-eservice-operations-interop-v1.yaml (hunk 2 above)
Comment (galales) [reply docs/openapi/api-eservice-operations-interop-v1.yaml:743]: the creation is in this PR, I don't see the point in doing and undoing it

Comment (AlessandroColucci) [reply docs/openapi/api-eservice-operations-interop-v1.yaml:743]: ok

**Review by galales — APPROVED** [review]

**Review by nttdata-rtorsoli — APPROVED** [review]

**Pull request merged**